test/testers.jl

```diff
@@ -241,10 +209,44 @@ end
     end
 
     @testset "TestIterator input" begin
+        function iterfun(iter)
+            state = iterate(iter)
+            state === nothing && error()
+            (x, i) = state
+            s = x^2
+            while true
+                state = iterate(iter, i)
+                state === nothing && break
+                (x, i) = state
+                s += x^2
+            end
+            return s
+        end
+
+        function ChainRulesCore.frule((_, Δiter), ::typeof(iterfun), iter)
+            iter_Δiter = zip(iter, Δiter)
+            state = iterate(iter_Δiter)
+            state === nothing && error()
+            # for some reason the following line errors if the frule is defined within a testset
+            ((x, Δx), i) = state
+            return iterfun(iter), sum(2 .* iter.data .* Δiter.data)
+            s = x^2
+            ∂s = 2 * x * Δx
+            while true
+                state = iterate(iter_Δiter, i)
+                state === nothing && break
+                ((x, Δx), i) = state
+                s += x^2
+                ∂s += 2 * x * Δx
+            end
+            return s, ∂s
+        end
+
         function ChainRulesCore.rrule(::typeof(iterfun), iter::TestIterator)
             function iterfun_pullback(Δs)
                 data = iter.data
                 ∂data = (2 * Δs) .* conj.(data)
+                @show ∂data
```

Comment:
do we still want this `@show`? (it wasn't in the code before)

Reply:
We do not
```suggestion
```